lib/miq_concurrency.rb

```diff
@@ -0,0 +1,46 @@
+module MiqConcurrency
+  module PGMutex
+    class << self
+      require 'zlib'
+      require 'securerandom'
+
+      def try_advisory_lock(lock_name)
+        execute_successful?('pg_try_advisory_lock', lock_name)
+      end
+
+      def release_advisory_lock(lock_name)
+        execute_successful?('pg_advisory_unlock', lock_name)
+      end
+
+      def count_advisory_lock(lock_name)
+        establish_connection
+        hashcode = stable_hashcode(lock_name)
+        sql = "SELECT count(*) from pg_locks "\
+              "WHERE locktype = 'advisory' "\
+              "AND objid = #{hashcode} "\
+              "AND pid = pg_backend_pid()"
+        @connection.select_value(sql).to_i
+      end
+
+      private
+
+      def establish_connection
+        unless ActiveRecord::Base.connected?
+          ActiveRecord::Base.establish_connection(ActiveRecord::Base.configurations[Rails.env])
```

Comment:
Is this really needed? I believe we should be able to just call `AR::Base.connection` and it'll do the right thing.